module/TaskManagement/src/TaskManagement/Controller/TasksController.php

```diff
@@ -93,15 +101,38 @@ public function getList()
 		
 		$streamID = $this->getRequest()->getQuery('streamID');
 
-		$validator = new ValidatorChain();
-		$validator->attach(new Int())
+		$integerValidator = new ValidatorChain();
+		$integerValidator->attach(new Int())
 			->attach(new GreaterThan(['min' => 0, 'inclusive' => false]));
+		$dateValidator = new DateValidator();
+		$uuidValidator = new UserIdValidator(array('pattern' => '([a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12})'));
+		$emailValidator = new EmailAddressValidator(array('useDomainCheck' => false));
 		
-		$offset = $validator->isValid($this->getRequest()->getQuery("offset")) ? intval($this->getRequest()->getQuery("offset")) : 0;
-		$limit = $validator->isValid($this->getRequest()->getQuery("limit")) ? intval($this->getRequest()->getQuery("limit")) : $this->getListLimit(); 
-		
-		$totalTasks = $this->taskService->countOrganizationTasks($this->organization);
-		$availableTasks = is_null($streamID) ? $this->taskService->findTasks($this->organization, $offset, $limit) : $this->taskService->findStreamTasks($streamID, $offset, $limit);
+		$offset = $integerValidator->isValid($this->getRequest()->getQuery("offset")) ? intval($this->getRequest()->getQuery("offset")) : 0;
+		$limit = $integerValidator->isValid($this->getRequest()->getQuery("limit")) ? intval($this->getRequest()->getQuery("limit")) : $this->getListLimit();
+		$startOn = null;
+		$endOn = null;
+		if($dateValidator->isValid($this->getRequest()->getQuery("endOn"))){
+			$endOn = \DateTime::createFromFormat($dateValidator->getFormat(), $this->getRequest()->getQuery("endOn"));
+		}
+		if($dateValidator->isValid($this->getRequest()->getQuery("startOn"))){
+			$startOn = \DateTime::createFromFormat($dateValidator->getFormat(), $this->getRequest()->getQuery("startOn"));
+		}else if($endOn instanceof \DateTime){
+			$startOn = $this->getDefaultStartOn($endOn);
+		}
+		$memberId = $uuidValidator->isValid($this->getRequest()->getQuery("memberId")) ? $this->getRequest()->getQuery("memberId") : null;
+
+		if($emailValidator->isValid($this->getRequest()->getQuery("memberEmail"))){
+			$member = $this->userService->findUserByEmail($this->getRequest()->getQuery("memberEmail"));
+			if($member != null){
+				$memberId = $member->getId();
+			}else{
+				$memberId = "";

```

Comment:
Per fare in modo che la ricerca non produca nessun risultato quando si inserisce un'email non presente in database: se `$memberId` è `null` non si filtra per questo valore